src/mpl/src/gavl/mpl_gavl.c

```diff
@@ -164,82 +195,66 @@ static int gavl_intersect_cmp_func(uintptr_t ustart, uintptr_t len, gavl_tree_no
 int MPL_gavl_tree_create(void (*free_fn) (void *), MPL_gavl_tree_t * gavl_tree)
 {
     int mpl_err = MPL_SUCCESS;
-    gavl_tree_s *gavl_tree_iptr;
+    gavl_tree_s *tree_ptr;
 
-    gavl_tree_iptr = (gavl_tree_s *) MPL_malloc(sizeof(gavl_tree_s), MPL_MEM_OTHER);
-    if (gavl_tree_iptr == NULL) {
+    tree_ptr = (gavl_tree_s *) MPL_malloc(sizeof(gavl_tree_s), MPL_MEM_OTHER);
+    if (tree_ptr == NULL) {
         mpl_err = MPL_ERR_SHM_NOMEM;
         goto fn_fail;
     }
 
-    gavl_tree_iptr->root = NULL;
-    gavl_tree_iptr->gavl_free_fn = free_fn;
-    *gavl_tree = (MPL_gavl_tree_t) gavl_tree_iptr;
+    tree_ptr->root = NULL;
+    tree_ptr->gavl_free_fn = free_fn;
+    TREE_CLEAR_STACK(tree_ptr);
+    *gavl_tree = (MPL_gavl_tree_t) tree_ptr;
 
   fn_exit:
     return mpl_err;
   fn_fail:
     goto fn_exit;
 }
 
-int MPL_gavl_tree_insert(MPL_gavl_tree_t gavl_tree, const void *addr, uintptr_t len,
-                         const void *val)
+static void gavl_tree_search_node(gavl_tree_s * tree_ptr, uintptr_t addr, uintptr_t len,
```

Comment:
The `_node` suffix doesn't really add any value. The function returns something, and its argument signature already says what it will return. How about `gavl_tree_search_internal`, since you want to emphasize this is the internal search?

Reply:
sounds good